lib/sync/storage/dataset-clients.js

```diff
@@ -252,7 +252,9 @@ function doReadDatasetClientWithRecordsUseCache(datasetClientId, callback) {
         if (err) {
           return callback(err);
         }
-        cacheClient.set(cacheKey, JSON.stringify(results));
+        if (results) {
+          cacheClient.set(cacheKey, JSON.stringify(results));
```

Comment:
the caller will have to handle that. It's like when you do db.find() - it could return null if it can't find the data.

The error should be only used if the request can't be completed.